boot/assets.go

```diff
@@ -634,8 +642,9 @@ func (o *TrustedAssetsUpdateObserver) Canceled() error {
 		return fmt.Errorf("cannot write modeeenv: %v", err)
 	}
 
-	// TODO:UC20:
-	// - reseal with a given state of modeenv
+	if err := resealKeyToModeenv(o.model, o.modeenv); err != nil {
+		return fmt.Errorf("cannot reseal encryption key after a canceled update: %v", err)
```

Comment:
we do it rarely but for same reasons we might have to do this like:

"while canceling gadget update: %v"